compiler/rustc_middle/src/mir/query.rs

```diff
@@ -38,10 +38,91 @@ pub enum UnsafetyViolationDetails {
     AccessToUnionField,
     MutationOfLayoutConstrainedField,
     BorrowOfLayoutConstrainedField,
-    CallToFunctionWith,
+    CallToFunctionWith { func_did: DefId, body_did_def_id: DefId },
 }
 
 impl UnsafetyViolationDetails {
+    fn missing_features(&self, tcx: TyCtxt<'_>) -> Vec<Symbol> {
```

Comment:
Since we know that there must be a missing feature if this error occurs, you can avoid the `match self` in the other two functions by just checking that the vec is not empty.